### DISCUSSION TITLE
createBrowserRouter loader request is empty

### DISCUSSION THREAD

**Discussion by j-waaang** [remix-run/react-router#11253] [Q&A]
The request object of my loader is always empty. Anyone knows whats up?

My router:
```
const router = createBrowserRouter(routes, routerOptions);

export default function Router() {
  return <RouterProvider router={router} />;
}
```

My route:
```
{
    element: <Page />,
    path: path,
    loader: pageloader,
  }
```

My loader:
```
export function routerloader(props: LoaderFunctionArgs) {
  console.log(JSON.stringify(props, null, 2)); }
```

The log:
```
{
  "request": {},
  "params": {}
}
```

**Comment by brophdawg11** [comment]
Request's are not JSON stringifiable - you will need to print out the request itself, or specific fields to see the internals:

![Screenshot 2024-02-05 at 1 02 20 PM](https://github.com/remix-run/react-router/assets/1609022/4117b62a-e29d-40e0-8b1d-63bd6cb32e32)
